### PR TITLE
Implemented Validation and Default behavior (#12)

### PR DESCRIPTION
**Pull request by ellis-bigelow**
Webhook server to follow. We will be able to upgrade to 
```mgr.GetWebhookServer().Register("/foo", hook)``` in 1.13. 
This implementation will not need to change.

/assign @yuzisun 
/assign @cliveseldon

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kfserving/21)

### CODE REVIEW

**Comment by ellis-bigelow** [comment]
Strange -- somehow @yuzisun is the author. I think this is due to a rebase and commit squash. Let me fix this.

File: pkg/apis/serving/v1alpha1/kfservice_types.go [hunk 1]
```diff
@@ -21,10 +18,23 @@ import (
...
+		},
+	},
+}
+
 // KFServiceSpec defines the desired state of KFService
 type KFServiceSpec struct {
-	MinReplicas int32 `json:"minReplicas"`
-	MaxReplicas int32 `json:"maxReplicas"`
+	MinReplicas int `json:"minReplicas"`
+	MaxReplicas int `json:"maxReplicas"`
```
Comment (yuzisun) [inline pkg/apis/serving/v1alpha1/kfservice_types.go]: Any reason why min/max replicas are required fields? Looks in order to have auto scale to zero user has to explicit set min=max=0?

Comment (ellis-bigelow) [reply pkg/apis/serving/v1alpha1/kfservice_types.go]: No. These should be optional. Let's cleanup in another.

File: pkg/apis/serving/v1alpha1/kfservice_defaults.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+// DefaultXGBoostVersion runs if not provided in XGBoostSpec
+var DefaultXGBoostVersion = "1.12"
+
+// DefaultScikitLearnVersion runs if not provided in ScikitLearnSpec
+var DefaultScikitLearnVersion = "1.12"
+
+// DefaultModelServerResourceRequirements sets initial limits on serving pods
+var DefaultModelServerResourceRequirements = v1.ResourceRequirements{
+	Requests: v1.ResourceList{
+		"cpu": resource.MustParse("1"),
```
[the remark this answers is not in the document]
Comment (ellis-bigelow) [reply pkg/apis/serving/v1alpha1/kfservice_defaults.go]: Yes probably. We should do a pass at some point to pull all of this out. Are you familiar with setting that up?

[the remark this answers is not in the document]
Comment (yuzisun) [reply pkg/apis/serving/v1alpha1/kfservice_defaults.go]: ye, we have other stuff which needs to move to ConfigMap, I will take care of this.

**Pull request merged**